social/backends/open_id.py

```diff
@@ -360,3 +437,16 @@ def request_access_token(self, *args, **kwargs):
         response = self.get_json(*args, **kwargs)
         self.id_token = self.validate_and_return_id_token(response['id_token'])
         return response
+
+    def user_data(self, access_token, *args, **kwargs):
+        return self.get_json(self.USERINFO_URL,
+                             headers={'Authorization': 'Bearer {0}'.format(access_token)})
+
+    def get_user_details(self, response):
+        return {
+            'username': response.get('preferred_username'),
+            'email': response.get('email'),
+            'fullname': response.get('name'),
+            'first_name': response.get('given_name'),
+            'last_name': response.get('family_name'),
+        }
```

Comment:
I appreciate your suggestion, but these fields have been standardized for OIDC in http://openid.net/specs/openid-connect-core-1_0.html#rfc.section.5.1. I therefore would prefer to use these as a sensible default. I guess most users would override the OpenIdConnectAuth to customize this for the actual values the backend provides (if the backend doesn't adhere to the RFC).